gaps/gap_node_descriptor/gap_node_descriptor.md

```diff
@@ -0,0 +1,104 @@
+
+## Specification
+
+Building on [Golem certificate structure](https://github.com/golemfactory/golem-architecture/blob/master/gaps/gap-25_golem_certificates/gap-25_golem_certificates.md) the Node descriptors are also utilizing JSON representation. This allows us to reuse previously defined data structures and create a uniform environment for digital signatures.
+
+Node descriptors are the first things to utilize Golem certificates and allow a verification of identity up to a trusted party. At the same time this chain of trust also enables restricting network permissions at each step, allowing the creation of node descriptors with the minimum required permissions in a given context.
+
+This GAP introduces the key usage `signNode` for [Golem certificates](https://github.com/golemfactory/golem-architecture/blob/master/gaps/gap-25_golem_certificates/gap-25_golem_certificates.md#key-usage).
+
+### Schema
+
+The schema heavily relies on references to previously defined items in [GAP-25](https://github.com/golemfactory/golem-architecture/blob/master/gaps/gap-25_golem_certificates/gap-25_golem_certificates.md). When a node descriptor is signed, all data in the `nodeDescriptor` property are digitally signed to add cryptographic integrity protection. Similarly to the previously defined certificate schema, the node descriptor schema does not restrict additional properties allowing adding contextual information and allowing compatibility with future additions.
+
+The schema file can be found in the `schemas/v1` folder and accessible at `https://golem.network/schemas/v1/node-descriptor.schema.json`.
+
+#### Node Id
+
+The `nodeId` property contains the node's identifier on the network in hexadecimal format.
+
+#### Permissions
+
+Details of how the node is permitted to use the Golem network. The content of this property is governed by the [permissions schema](https://golem.network/schemas/v1/permissions.schema.json).
+
+#### Validity Period
+
+The time period for which this node descriptor is valid. Semantics defined in the [certificate schema](https://github.com/golemfactory/golem-architecture/blob/master/gaps/gap-25_golem_certificates/gap-25_golem_certificates.md#validity-period).
+
+#### Signature
+
+Semantics of the signature are defined in the [certificate schema](https://github.com/golemfactory/golem-architecture/blob/master/gaps/gap-25_golem_certificates/gap-25_golem_certificates.md#signature) with one difference. A node descriptor cannot be used to create signatures so it cannot be self signed.
+
+### Signature creation and verification
+
+The process is the same as described for [certificates](https://github.com/golemfactory/golem-architecture/blob/master/gaps/gap-25_golem_certificates/gap-25_golem_certificates.md#signature-creation-and-verification), but the signature is created for data in the `nodeDescriptor` property instead of the `certificate` one.
```

Comment:
Oslo, `key-usage: "signNode"` in signer must be verified

Reply:
I have it in the details in the next two paragraphs. I changed the wording from `same` to `quite similar`.